Assets/UPM/Console/CommandDiscovery.cs

```diff
@@ -55,15 +68,30 @@ public void Invoke(string methodName)
 
             Execute(commandMap[splitString[0]].Program, argumentList);
         }
+        else
+        {
+            Debug.Log("Cant find Command: " + splitString[0]);
+        }
     }
 
     private IEnumerable<string> GenerateArgumentList(IEnumerable<string> arguments)
     {
         return arguments;
     }
 
-    private void Execute(MethodInfo program, IEnumerable<string> arguments)
+    private void Execute(MethodInfo program, IEnumerable<string> arguments, Type activator = null)
     {
-        program.Invoke(null, arguments.ToArray());
+        if (program.IsStatic)
+        {
+            program.Invoke(null, arguments.ToArray());
+        }
+        else
+        {
+            //TODO fix non Static, because then we need to execute with the class that has this Methode/Program
+            object instance = Activator.CreateInstance(activator); // activator = Type = the class/Type that has the Methode ?
```

Comment:
But what scenario exists where you'd want a non-static command. I can't think of any off the top of my head. I can't think of a scenario where you'd write `SomeGameObject.Translate-Position (0,0,0)`. I picture that as `Find 'Some Game Object' > Translate-Position (0,0,0)` Is there something I'm not considering?